### ISSUE TITLE
NavNavigationFile (.nav2) Format [$1013.00] [Paid Out]

### ISSUE THREAD

**Issue by morbidslinky** [MGSV-Modding/research-bounties#1]
**1/3/21 Notice:**
The [submission from OldBanana12](https://github.com/oldbanana12/Nav2) has been accepted. The bounty has been paid out.

**8/17/20 Notice:**
A user is in the process of submitting their solution to this bounty. If their submission is accepted, they will receive the bounty, undivided, and this issue will be closed.
If you have made substantial progress on this bounty independently from this user's solution, please make yourself known. I would like to ensure that compensation is awarded fairly if there are more than one solutions on the cusp of submission.

**The file format or specific topic to be researched:**

NavNavigationFile (.nav2)

**The specific game(s) associated with the file:**

Metal Gear Solid V: Ground Zeroes (2014)
Metal Gear Solid V: The Phantom Pain (2015)
Metal Gear Survive (2018)

**A summary of the file format's purpose:**

The NavNavigationFile is associated with, and believed to be chiefly responsible for, "navmesh" NPC map navigation. 

**Any format documentation or research currently available:**

- A partially completed Binary Template is available [here](https://github.com/MGSV-Modding/research-bounties/files/4749888/nav2.bt.zip).
- A partially completed Nav2 Parsing Tool (C#) can be found [here](https://github.com/BobDoleOwndU/Nav2Parser).
- Additional research discussion, resources and support can be gathered through the [Modding Community Discord](https://discord.gg/qyVQQs2).

**Acceptance Criteria:**

An acceptable solution must satisfy the following:
- Completely resolve the structure of the format, as well as the purpose of all the format's unknown fields.
- Deliver a tool which can reliably: 
  - Interpret a NavNavigationFile of arbitrary origin* and size, and deserialize it into a data-serialization language file (XML Preferred).
  - Interpret a structurally correct, theoretically functional, and deserialized NavNavigationFile (formatted in the same data-serialization language as before), and serialize it into a verifiably functional* NavNavigationFile file.

*\*To the extent that the NavNavigationFile is compatible with Metal Gear Solid V: The Phantom Pain. It is not required for the solution to provide compatibility with Ground Zeroes or Survive.*

https://www.bountysource.com/issues/91675424-navnavigationfile-nav2-format

**Comment by morbidslinky** [comment]
**8/17/20 Notice:**
A user is in the process of submitting their solution to this bounty. If their submission is accepted, they will receive the bounty, undivided, and this issue will be closed.
If you have made substantial progress on this bounty independently from this user's solution, please make yourself known. I would like to ensure that compensation is awarded fairly if there are more than one solutions on the cusp of submission.

**Comment by oldbanana12** [comment]
I have published the information I have here: https://github.com/oldbanana12/Nav2

This includes an almost complete .bt file along with some high level descriptions of the file format, code that can parse .nav2 files and extract and display the data structures within in a meaningful way, and some code that can take a Wavefront .obj mesh and turn it into a .nav2 file.

Currently it's not possible to do a round trip from a .nav2 to .obj back to .nav2 with my tools due to the complexity of the format. Though it may work for some of the simpler maps if you do some manual processing of the .obj mesh in the middle. That said, it should be possible to make new .obj files for maps that do not have .nav2 files and create working AI navigation as a result.

I appreciate that this does not 100% meet the brief, but I no longer have the time to work on this. As such, I feel it's better to release the information I have rather than sit on it in the hope that it proves useful and someone else can pick it up. I'm going to submit a claim for the bounty on Bountysource, but understand if you're unwilling to approve it given that it doesn't meet the brief.

Edit: Turns out I can't submit a claim for the bounty until this issue is marked as closed on GitHub.
